openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -115,30 +117,43 @@ public String getPatientIdentifierTypeUuidByName(Config config, String identifie
     }
 
     @Override
+    public List<Identifier> getPatientIdentifierList(Config config, String patientUuid) {
+        String responseJson = getJson(config, "/patient/{patientUuid}/identifier?v=full", patientUuid);
+        return ((IdentifierListResult) JsonUtils.readJson(responseJson, IdentifierListResult.class)).getResults();
+    }
+
+    @Override
     public void deletePatient(Config config, String uuid) {
         delete(config, "/patient/{uuid}?purge", uuid);
     }
 
     @Override
     public void updatePatientMotechId(Config config, String patientUuid, String newMotechId) {
-        Identifier patientIdentifier = getPatientIdentifier(config, patientUuid);
-        patientIdentifier.setIdentifier(newMotechId);
+        List<Identifier> patientIdentifiers = getPatientIdentifierList(config, patientUuid);
+        for (Identifier patientIdentifier : patientIdentifiers) {
+            if ("MOTECH Id".equals(patientIdentifier.getIdentifierType().getName())) {
+                patientIdentifier.setIdentifier(newMotechId);
+
+                String requestJson = buildGson().toJson(patientIdentifier);
+                postForJson(config, requestJson, "/patient/{patientUuid}/identifier/{identifierUuid}", patientUuid,
+                        patientIdentifier.getUuid());
+                break;
+            }
+        }
+    }
 
-        String requestJson = buildGson().toJson(patientIdentifier);
+    @Override
+    public void updatePatientIdentifier(Config config, String patientUuid, Identifier updatedIdentifier) {
```

Comment:
change the name of variable, this identfier will be updated, just name this 'identifier'